src/System.IO.FileSystem.Watcher/src/System/IO/FileSystemWatcher.Linux.cs

```diff
@@ -299,7 +298,10 @@ private WatchedDirectory AddDirectoryWatchUnlocked(WatchedDirectory parent, stri
 
                 // Add a watch for the full path.  If the path is already being watched, this will return 
                 // the existing descriptor.  This works even in the case of a rename.
-                int wd = (int)SysCall(fd => Interop.libc.inotify_add_watch(fd, fullPath, (uint)_notifyFilters));
+                int wd = (int)SysCall(
+                    (fd, fullpath, runningInstance) => Interop.libc.inotify_add_watch(fd, fullpath, (uint)runningInstance._notifyFilters),
```

Comment:
Maybe ```thisRef``` instead of ```runningInstance```, just to highlight that we're really trying to access ```this``` but just using a different name for it?